### ISSUE TITLE
license function via API?

### ISSUE THREAD

**Issue by ghost** [z/xonotic-map-repository#13]
Would it be possible to provide a function to show the license(s) of a map?

I'm thinking of possible inclusion in operating systems where it makes sense to know this kind of information, what can be included and what not.

The data is already in the big csv file, but I think it would be easier if the possibility were given to get information for a specific map without the webbrowser or downloading a big csv first.

**Comment by z** [comment]
Yes, it's possible, but the major issue we have is the lack of declared licenses for maps.

Current coverage is roughly 2% of maps in the repository that specify a license in a standard way.

Moving towards a "registry", we can require all submitted maps specify a license.

**Comment by ghost** [comment]
Okay.
But no specified license already implies a license (unless this is different for map creations, but "creation belongs to the artist" should still apply). To change this license is then "only" a matter of assembling a team of people to handle the license questions to all map creators. Some can even be fixed without contacting people, like the original Quake maps I seem to have noticed there.

If no redistribution is implied in the original licenses, then it will be "fun" times to fix this as you seem to offer some (all?) maps via your domain.

**Comment by z** [comment]
> But no specified license already implies a license (unless this is different for map creations, but "creation belongs to the artist" should still apply).

There is an issue with assuming the creation belongs to the artist, as an artifact of the modding community, many maps include assets which the map creator does not have a license for.  Furthermore, I've been told that some mappers will purposely not include a license, to discourage reuse, or imply that their package includes questionably licensed assets.  I'm not sure how to deal with those edge case, other than say the license is "unspecified", and let the consumer decide.

I'm of course in favor of clearer licensing, this is why I built it into the data model in the first place.  Unfortunately, there is no silver bullet.

The original version of this project was a package analysis.  Many of these packages have been floating around since the 90s, just barely compatible with Xonotic.  I would like to at least better compartmentalize these packages, but there is no silver bullet for that either, due to how the engine downloads maps.

**Comment by ghost** [comment]
Interesting that it's still like in the late 90s / early 2000s. I've worked on maps then and wasn't aware of copy-left licenses, so I did not use any license.

Maybe you are right and "unspecified" is the best way to deal with this.
I will get some feedback from one or two projects I'm in on how maps and game extending data in general is treated/labeled.
